src/test/java/roomescape/client/PaymentClientTest.java

```diff
@@ -0,0 +1,81 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpMethod;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.web.client.RestClient;
+import org.springframework.http.MediaType;
+import roomescape.config.RestClientConfiguration;
+import roomescape.dto.reservation.TossPaymentConfirmRequestDto;
+import roomescape.exception.PaymentConfirmClientException;
+import roomescape.exception.PaymentConfirmServerException;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.method;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withBadRequest;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withResourceNotFound;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {RestClientConfiguration.class, ObjectMapper.class})
+public class PaymentClientTest {
+
+    @Autowired
+    private RestClient.Builder restClientBuilder;
+
+    private PaymentClient paymentClient;
+
+    private MockRestServiceServer mockRestServiceServer;
+
+    @BeforeEach
+    void setUp() {
+        mockRestServiceServer = MockRestServiceServer.bindTo(restClientBuilder).build();
+        RestClient testRestClient = restClientBuilder.build();
+        paymentClient = new PaymentClient(testRestClient);
+    }
+
+    @DisplayName("잘못된 시크릿키로 결제 승인 요청시 500에러가 반환된다.")
+    @Test
+    void paymentExceptionTest() {
+
+        mockRestServiceServer.expect(requestTo("https://api.tosspayments.com/v1/payments/confirm"))
+                .andExpect(method(HttpMethod.POST))
+                .andRespond(withBadRequest()
+                        .body("{\"code\":\"INVALID_API_KEY\",\"message\":\"잘못된 시크릿키 연동 정보 입니다.\"}")
+                        .contentType(MediaType.APPLICATION_JSON));
+
+
+        assertThatThrownBy(
+                () -> paymentClient.confirmPayment(new TossPaymentConfirmRequestDto(
+                        "paymentKey", "orderId", 1000L
+                ))
+        ).isInstanceOf(PaymentConfirmServerException.class);
+
+        mockRestServiceServer.verify();
+    }
+
+    @DisplayName("paymentkey를 클라이언트에서 획득하지 않은 값으로 요청 시 400에러가 반환된다.")
+    @Test
+    void invalidPaymentKeyExceptionTest() {
+
+        mockRestServiceServer.expect(requestTo("https://api.tosspayments.com/v1/payments/confirm"))
+                .andExpect(method(HttpMethod.POST))
+                .andRespond(withResourceNotFound()
+                        .body("{\"code\":\"NOT_FOUND_PAYMENT_SESSION\",\"message\":\"결제 시간이 만료되어 결제 진행 데이터가 존재하지 않습니다.\"}")
```

Comment:
```suggestion
                        .body("""
                                {
                                    "code":"NOT_FOUND_PAYMENT_SESSION",
                                    "message":"결제 시간이 만료되어 결제 진행 데이터가 존재하지 않습니다."
                                }
                        """)
```
JSON 문자열이네요! 읽기 좋게 text block 처리해보면 어떨까요?

Reply:
오오 이렇게 가독성을 올릴 수 있군요! 반영하여 수정했습니다 감사합니다 👍

[test: JSON 문자열 가독성 개선](https://github.com/woowacourse/spring-roomescape-payment/pull/223/commits/161ffe261e9285440b59da57d7f5ff53893d21c5)